scripts/troubleshooting/hs-manifest-gather/script.sh

```diff
@@ -0,0 +1,132 @@
+    echo "yq binary not found in PATH"
+    exit 1
+  fi
+
+  # Sanity check to make sure desired cluster namespace exists
+  hcp_ns_list_length=$(oc get namespace -l api.openshift.com/id="${CLUSTER_ID}" -o json | jq -r '.items | length')
+  if [ "${hcp_ns_list_length}" -ne 1 ]; then
+    echo "Number of HCP namespaces matching ${CLUSTER_ID} must be 1, ${hcp_ns_list_length} found"
+    exit 1
+  fi
+
+  # Fetch the target HCP namespace for the cluster name
+  hc_namespace=$(oc get namespace -l api.openshift.com/id="${CLUSTER_ID}" -o json | jq -r '.items[0].metadata.labels["kubernetes.io/metadata.name"]')
+
+  # Fetch the HostedCluster name
+  hc_name=$(oc get HostedCluster.hypershift.openshift.io -l api.openshift.com/id="${CLUSTER_ID}" -n "${hc_namespace}" -o json | jq -r '.items[0].metadata.name')
+
+  
+  # TODO(ACM-16170): replace this with an official ACM release image once it's available
+  # local acm_hypershift_image="registry.redhat.io/rhacm2/acm-must-gather-rhel9:v2.11"
+  local acm_hypershift_image="quay.io/rokejungrh/must-gather:v2.13.0-33-linux"
+
+  # Create gather script command
+  local gather_script="/usr/bin/gather hosted-cluster-namespace=${hc_namespace} hosted-cluster-name=${hc_name}"
+  
+  echo "Collecting ACM HyperShift must-gather(manifests only) for cluster ${hc_name} in namespace ${hc_namespace}"
+  
+  # Execute must-gather
+  if ! oc adm must-gather --dest-dir="${DUMP_DIR}" --image="${acm_hypershift_image}" -- "${gather_script}"; then
+    echo "failed to gather ACM HyperShift manifests data: oc adm must-gather failed"
+    return 1
+  fi
+  
+  echo "ACM HyperShift must-gather(manifests only) completed successfully"
+  ls -alh "${DUMP_DIR}"
+  return 0
+}
+
+# Function to remove files with Secrets and CERTIFICATE data
+remove_sensitive_files() {
+  cd "${DUMP_DIR}"
+
+  find . -type f -name "*.yaml" -print0 | while IFS= read -r -d '' file; do
+    if [ "$(yq e '.kind == "Secret"' "${file}" 2> /dev/null)" = "true" ]; then
+      echo "Removing ${file} because it contains a Secret"
+      rm -f "${file}"
+    elif [ "$(yq e '.kind == "SecretList"' "${file}" 2> /dev/null)" = "true" ]; then
+      echo "Removing ${file} because it contains a SecretList"
+      rm -f "${file}"
+    elif [ "$(yq e 'select(.data[] | contains("CERTIFICATE")) | [.] | length > 0' "${file}" 2> /dev/null)" = "true" ]; then
+      echo "Removing ${file} because it contains CERTIFICATE data"
+      rm -f "${file}"
+    elif [ "$(yq e 'select(.items[].data[] | contains("CERTIFICATE")) | [.] | length > 0' "${file}" 2> /dev/null)" = "true" ]; then
+      echo "Removing ${file} because it contains CERTIFICATE data"
+      rm -f "${file}"
+    fi
+  done
+
+  return 0
+}
+
+# Function to compress the dump into a tarball
+create_tarball() {
+  cd "${DUMP_DIR}"
+
+  if [ -f "${TARBALL_PATH}" ]; then
+    echo "Tarball ${TARBALL_PATH} already exists. Exiting."
+    exit 0
+  fi
+
+  # Compress the dump directory
+  tar -czvf "${TARBALL_PATH}" ./*
```

Comment:
Looks the tarball is written inside `${DUMP_DIR}`, will it including itself if rerun to compress the  dump again?